service/history/handler.go

```diff
@@ -53,9 +53,14 @@ import (
 	"github.com/uber/cadence/service/history/resource"
 	"github.com/uber/cadence/service/history/shard"
 	"github.com/uber/cadence/service/history/task"
+	"github.com/uber/cadence/service/history/workflowcache"
 )
 
-const shardOwnershipTransferDelay = 5 * time.Second
+const (
+	shardOwnershipTransferDelay = 5 * time.Second
+	workflowIDCacheTTL          = 1 * time.Second
+	workflowIDCacheMaxCount     = 10_000
```

Comment:
[um what? Go supports underscores in numbers?](https://go.dev/ref/spec#:~:text=Integer%20literals,-An%20integer%20literal&text=In%20hexadecimal%20literals%2C%20letters%20a,not%20change%20the%20literal's%20value.) I never new that

Reply:
Yep, [introduced in 1.13](https://go.dev/doc/go1.13#language). Though honestly I was not consciously aware (just wrote it), if you had asked me yesterday, I'd not have known without trying it out 😅.